src/components/shareDiv.vue

```diff
@@ -144,8 +148,8 @@ export default {
 	methods: {
 		removeShare(item) {
 			// Filter out the removed item
-			const users = this.userShares.filter(user => !(user.id === item.id && !item.isGroup))
-			const groups = this.groupShares.filter(group => !(group.id === item.id && item.isGroup))
+			const users = this.userShares.filter(user => user.shareWith !== item.shareWith && user.shareType !== item.shareType)
+			const groups = this.groupShares.filter(group => group.shareWith !== item.shareWith && group.shareType !== item.shareType)
```

Comment:
The old logic was much easier to understand with the inverted brackets. And as it is now, all shares of the same type will be deleted, when deleting one.
I'd say just use the old code, and check for the item type to be the expected one.
```suggestion
			const users = this.userShares.filter(user => !(user.shareWith === item.shareWith && item.shareType === this.SHARE_TYPES.SHARE_TYPE_USER))
			const groups = this.groupShares.filter(group => !(group.shareWith === item.shareWith && item.shareType === this.SHARE_TYPES.SHARE_TYPE_GROUP))
```

Reply:
> And as it is now, all shares of the same type will be deleted, when deleting one.

Ah, weird! Nice catch!